spring-cloud-gcp-storage/src/main/java/org/springframework/cloud/gcp/storage/GoogleStorageResourceObject.java

```diff
@@ -119,6 +125,42 @@ public Blob getGoogleStorageObject() throws IOException {
 		return this.storage.get(getBlobId());
 	}
 
+	/**
+	 * Creates a signed URL to an object if it exists.
+	 * @param timeUnit the time unit used to determine how long the URL is valid.
+	 * @param timePeriods the number of periods to determine how long the URL is valid.
+	 * @return the URL if the object exists, and null if it does not.
+	 * @throws IOException
+	 */
+	public URL createSignedUrl(TimeUnit timeUnit, long timePeriods) throws IOException {
+		Blob blob = this.getGoogleStorageObject();
+		if (blob == null) {
+			return null;
+		}
+		String privateKeyPath = getGcpPrivateKeyPath();
+		return this.storage.signUrl(
+				BlobInfo.newBuilder(blob.getBucket(), blob.getName()).build(),
+				timePeriods, timeUnit, getSignUrlOptionForPrivateKey(privateKeyPath));
```

Comment:
Why are you passing credentials here at all?
`this.storage` already contains the correct credentials, no need to pass them here again.

Plus, if for whatever reason they needed to be passed here again, you would depend on `GoogleCredentials`, not get the credentials on your own.

Reply:
What's needed is a ServiceAccountCredentials, which is a child of GoogleCredentials, and I'll try looking for if that is available at all, but it might not be, since none of the existing code requires specifically a service account credentials to work.

Reply:
I see what you mean.

I think the best way forward here is to add a `GoogleCredentials` argument to this method and then checking if it's an instance of `ServiceAccountCredentials` and, if so, cast it.

If it isn't, we should fail with clear instructions for the user (e.g., `createSignedUrl` only works if you're using service account creds).

Reply:
Makes sense, will do.